### ISSUE TITLE
``UndefError`` in printing JuMPArray with index set of tuples

### ISSUE THREAD

**Issue by joehuchette** [JuliaOpt/JuMP.jl#922]
```jl
using JuMP
m = Model()
JuMP.@variable(m, [(0,0)])
```
```
Error showing value of type JuMP.JuMPArray{JuMP.Variable,1,Tuple{Tuple{Int64,Int64}}}:
ERROR: UndefRefError: access to undefined reference
 in first(::JuMP.ValueIterator{JuMP.Variable,1}) at ./abstractarray.jl:134
 in cont_str(::Type{T}, ::JuMP.JuMPArray{JuMP.Variable,1,Tuple{Tuple{Int64,Int64}}}, ::Dict{Symbol,String}) at /Users/huchette/.julia/v0.5/JuMP/src/print.jl:453
 in show at /Users/huchette/.julia/v0.5/JuMP/src/print.jl:385 [inlined]
 in show at ./replutil.jl:4 [inlined]
 in display(::Base.REPL.REPLDisplay{Base.REPL.LineEditREPL}, ::MIME{Symbol("text/plain")}, ::JuMP.JuMPArray{JuMP.Variable,1,Tuple{Tuple{Int64,Int64}}}) at ./REPL.jl:132
 in display(::Base.REPL.REPLDisplay{Base.REPL.LineEditREPL}, ::JuMP.JuMPArray{JuMP.Variable,1,Tuple{Tuple{Int64,Int64}}}) at ./REPL.jl:135
 in display(::JuMP.JuMPArray{JuMP.Variable,1,Tuple{Tuple{Int64,Int64}}}) at ./multimedia.jl:143
 in print_response(::Base.Terminals.TTYTerminal, ::Any, ::Void, ::Bool, ::Bool, ::Void) at ./REPL.jl:154
 in print_response(::Base.REPL.LineEditREPL, ::Any, ::Void, ::Bool, ::Bool) at ./REPL.jl:139
 in (::Base.REPL.##22#23{Bool,Base.REPL.##33#42{Base.REPL.LineEditREPL,Base.REPL.REPLHistoryProvider},Base.REPL.LineEditREPL,Base.LineEdit.Prompt})(::Base.LineEdit.MIState, ::Base.AbstractIOBuffer{Array{UInt8,1}}, ::Bool) at ./REPL.jl:652
 in run_interface(::Base.Terminals.TTYTerminal, ::Base.LineEdit.ModalInterface) at ./LineEdit.jl:1579
 in run_interface(::Base.Terminals.TTYTerminal, ::Base.LineEdit.ModalInterface) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in run_frontend(::Base.REPL.LineEditREPL, ::Base.REPL.REPLBackendRef) at ./REPL.jl:903
 in run_repl(::Base.REPL.LineEditREPL, ::Base.##930#931) at ./REPL.jl:188
 in _start() at ./client.jl:360
 in _start() at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
```

**Comment by joehuchette** [comment]
Potentially related:
```jl
using JuMP
m = Model()
@variable(m, [[1,1]])
```
```
Error showing value of type JuMP.JuMPArray{JuMP.Variable,1,Tuple{Array{Int64,1}}}:
ERROR: UndefRefError: access to undefined reference
 in first(::JuMP.ValueIterator{JuMP.Variable,1}) at ./abstractarray.jl:134
 in cont_str(::Type{T}, ::JuMP.JuMPArray{JuMP.Variable,1,Tuple{Array{Int64,1}}}, ::Dict{Symbol,String}) at /Users/huchette/.julia/v0.5/JuMP/src/print.jl:453
 in show at /Users/huchette/.julia/v0.5/JuMP/src/print.jl:385 [inlined]
 in show at ./replutil.jl:4 [inlined]
 in display(::Base.REPL.REPLDisplay{Base.REPL.LineEditREPL}, ::MIME{Symbol("text/plain")}, ::JuMP.JuMPArray{JuMP.Variable,1,Tuple{Array{Int64,1}}}) at ./REPL.jl:132
 in display(::Base.REPL.REPLDisplay{Base.REPL.LineEditREPL}, ::JuMP.JuMPArray{JuMP.Variable,1,Tuple{Array{Int64,1}}}) at ./REPL.jl:135
 in display(::JuMP.JuMPArray{JuMP.Variable,1,Tuple{Array{Int64,1}}}) at ./multimedia.jl:143
 in print_response(::Base.Terminals.TTYTerminal, ::Any, ::Void, ::Bool, ::Bool, ::Void) at ./REPL.jl:154
 in print_response(::Base.REPL.LineEditREPL, ::Any, ::Void, ::Bool, ::Bool) at ./REPL.jl:139
 in (::Base.REPL.##22#23{Bool,Base.REPL.##33#42{Base.REPL.LineEditREPL,Base.REPL.REPLHistoryProvider},Base.REPL.LineEditREPL,Base.LineEdit.Prompt})(::Base.LineEdit.MIState, ::Base.AbstractIOBuffer{Array{UInt8,1}}, ::Bool) at ./REPL.jl:652
 in run_interface(::Base.Terminals.TTYTerminal, ::Base.LineEdit.ModalInterface) at ./LineEdit.jl:1579
 in run_interface(::Base.Terminals.TTYTerminal, ::Base.LineEdit.ModalInterface) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in run_frontend(::Base.REPL.LineEditREPL, ::Base.REPL.REPLBackendRef) at ./REPL.jl:903
 in run_repl(::Base.REPL.LineEditREPL, ::Base.##930#931) at ./REPL.jl:188
 in _start() at ./client.jl:360
 in _start() at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
```

**Comment by mlubin** [comment]
The issue is we weren't checking for duplicate indices in sets. Fixed in #1102.
